### ISSUE TITLE
Hello your app simply doesn't work with me i followed your instructions (please help ) 

### ISSUE THREAD

**Issue by madadoux** [sochix/TLSharp#199]
// what i'm doing wrong ?? 

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TLSharp.Core;
namespace ConsoleApplication1
{
    class Program
    {
        
 

        public void a() { }
        static void Main(string[] args)
        {

            Console.WriteLine("hello");

            op o = new op();
            o.Send(); 

        }

        
       
    }

    public class op
    {
        private string phoneNumber = "+201016705870";
        public async void Send()
        {

            try
            {
                var store = new FileSessionStore();
                var client = new TelegramClient(store, "session",
                                              78648, "26ee7de17c4355b6f837895665ac8ece");
                await client.Connect();

                var hash = await client.SendCodeRequest(phoneNumber); var code = "22797"; //recieved code
                var user = await client.MakeAuth(phoneNumber, hash, code);

                var res = await client.ImportContactByPhoneNumber("+201147467203");

                await client.SendMessage(res.Value, "sfdfds");
            }

            catch (Exception e )
            {
                Console.WriteLine(e.Message); 
            }
        }
    }
}

**Comment by steavy29** [comment]
Hi. You provided just a sample of code. What happens when you run it? Does it crash? How? Be more specific, please, so that someone can help you.

**Comment by madadoux** [comment]
it runs fine no expections but no message received   i don't know whether it works or not 
also i have replace the "void" type with "Task" >> not working 
and removed + from phone number >> not working also 

even i put  print statment on the middle of code to test if it excutes  completely or not
 
var res = await client.ImportContactByPhoneNumber("+201147467203");
Console.WriteLine ("hi there " ) ; 
 await client.SendMessage(res.Value, "sfdfds"); 


and the console never output it 

the code meant to send  a single message to this number 
thank you for replying :)

**Comment by steavy29** [comment]
What about `o.Send().Wait();` ? :)

**Comment by madadoux** [comment]
THANK U NOW its working it shows some excptions  but it working :)

**Comment by madadoux** [comment]
please don't close this issue untill i finish dev [1-2]days

**Comment by madadoux** [comment]
now  i sent request code 7 times in raw by mistake  and the server appearing to blocking me [the programm freezed] so what i can do now to unblock my self ? thanks in advance

**Comment by madadoux** [comment]
public async Task<bool> Connect(bool reconnect = false)
        {
            if (_session.AuthKey == null || reconnect)
            {
                var result = await Authenticator.DoAuthentication(_transport);
                _session.AuthKey = result.AuthKey;
                _session.TimeOffset = result.TimeOffset;
            }

            _sender = new MtProtoSender(_transport, _session);

            if (!reconnect)
            {
                var request = new InitConnectionRequest(_apiId);

                await _sender.Send(request);
                await _sender.Receive(request); // here freezes  using debugger 

                dcOptions = request.ConfigConstructor.dc_options;
            }

            return true;
        }

**Comment by steavy29** [comment]
Then you should open a separate issue where you will describe your new problem,

**Comment by madadoux** [comment]
https://github.com/sochix/TLSharp/issues/200

**Comment by steavy29** [comment]
Please, close this issue.
